# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/nsIQuotaManagerService.idl
# Commit: a604c144191f
# Full Hash: a604c144191f5c3c65169c072d4e52a1e2c1c23f
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-09 09:46:37
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Change nsIQuotaManagerService::ClearStoragesForPrincipal to support clearing of entire origins only; r=dom-storage-reviewers,jari
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195505
# ==============================================================================

diff -r 281a926003fd -r a604c144191f dom/quota/nsIQuotaManagerService.idl
--- a/dom/quota/nsIQuotaManagerService.idl	Wed Oct 09 05:34:36 2024 +0000
+++ b/dom/quota/nsIQuotaManagerService.idl	Wed Oct 09 05:34:37 2024 +0000
@@ -252,18 +252,10 @@
    *        "persistent" is different than being "persisted" via persist() and
    *        is only for chrome principals.  See bug 1354500 for more info.
    *        In general, null is the right thing to pass here.
-   * @param aClientType
-   *        An optional string that tells what client type of storages
-   *        will be cleared.  If omitted (or void), all client types will be
-   *        cleared for the principal.  If a single client type is provided
-   *        from Client.h, then only that client's storage will be cleared.
-   *        If you want to clear multiple client types (but not all), then you
-   *        must call this method multiple times.
    */
   [must_use] nsIQuotaRequest
   clearStoragesForPrincipal(in nsIPrincipal aPrincipal,
-                            [optional] in ACString aPersistenceType,
-                            [optional] in AString aClientType);
+                            [optional] in ACString aPersistenceType);
 
   /**
    * Removes all storages stored for the given client. The files may not be